angular/core/services/lmo_url_service.coffee

```diff
@@ -22,20 +19,23 @@ angular.module('loomioApp').factory 'LmoUrlService', (AppConfig) ->
       @buildModelRoute('d', d.key, d.title, params, options)
 
     searchResult: (r, params = {}, options = {}) ->
-      @discussion(r, params, options)
-
-    proposal: (p, params = {}, options = {}) ->
-      @discussion p.discussion(), _.merge(params, {proposal: p.key})
-
-    comment: (c, params = {}, options = {}) ->
-      @discussion c.discussion(), _.merge(params, {comment: c.id})
+      @discussion r, params, options
 
     user: (u, params = {}, options = {}) ->
       @buildModelRoute('u', u.username, null, params, options)
 
+    proposal: (p, params = {}) ->
+      @route model: p.discussion(), action: "proposal/#{p.key}", params: params
+
+    comment: (c, params = {}) ->
+      @route model: c.discussion(), action: "comment/#{c.id}", params: params
+
     membership: (m, params = {}, options = {}) ->
       @route model: m.group(), action: 'memberships', params: params
 
+    membershipRequest: (mr, params = {}, options = {}) ->
```

Comment:
Instead of lmo-hreffing `/d/abc?comment=123`, (which breaks because the 'new' angular router can't handle query params in that way, we're pointing to `/d/abc/comment/123`, meaning we can pick up the value from $routeParams, rather than $location.search()